### PR TITLE
Improve singular matrix power performance

### LINKED ISSUE #11964: Singular matrix gives wrong result with negative powers
```python
In [3]: m = Matrix([[1, 1], [2, 2]])

In [4]: m**(-1)
---------------------------------------------------------------------------
ValueError: Matrix det == 0; not invertible.

In [5]: ;var N
Out[5]: N

In [6]: (m**N).subs(N, -1)
Out[6]: 
⎡zoo  zoo⎤
⎢        ⎥
⎣zoo  zoo⎦

In [7]: (m**N)
Out[7]: 
⎡     N    N       N    N⎤
⎢  2⋅0    3       0    3 ⎥
⎢  ──── + ──    - ── + ──⎥
⎢   3     3       3    3 ⎥
⎢                        ⎥
⎢     N      N   N      N⎥
⎢  2⋅0    2⋅3   0    2⋅3 ⎥
⎢- ──── + ────  ── + ────⎥
⎣   3      3    3     3  ⎦

```

### PR DESCRIPTION
**Pull request by Abdullahjavednesar**
Fixes #11964

### CODE REVIEW

**Comment by Abdullahjavednesar** [comment]
Ping @jksuom

File: sympy/matrices/matrices.py [hunk 1]
```diff
@@ -1593,7 +1602,7 @@ def __pow__(self, num):
                 except AttributeError:
```
Comment (jksuom) [inline sympy/matrices/matrices.py]: It seems that ValueError should be added here.

File: sympy/matrices/matrices.py [hunk 2]
```diff
@@ -1593,7 +1602,7 @@ def __pow__(self, num):
...
                     return self._matrix_pow_by_recursion(num)
```
Comment (jksuom) [inline sympy/matrices/matrices.py]: This could probably also be `pass`.

**Comment by jksuom** [comment]
This looks good to me now. Thanks!

**Pull request merged**